python/rikai/spark/functions/vision.py

```diff
@@ -234,3 +237,79 @@ def spectrogram_image(
     return Image.from_array(
         np.frombuffer(output, np.uint8).reshape([size, size, 3]), output_uri
     )
+
+
+@udf(returnType=ArrayType(SegmentType()))
+def video_segmentor(
+    video: Union[VideoStream, YouTubeVideo],
+    threshold: float = 0.3,
+    fps: int = 30,
+) -> list:
+    """Returns list of video Segments from VideoStream or YouTubeVideo by
+    thresholding sum-of-absolute differences.
+    https://github.com/FFmpeg/FFmpeg/blob/master/libavfilter/f_select.c
+    Parameters
+    ----------
+    video : Video
+      An video object, either YouTubeVideo or VideoStream.
+    threshold: float, default 0.3
+      Scene detection score, values in [0, 1]
+    fps: int, default 30
+      Video frame rate in hertz
+    Return
+    ------
+    List
+      Returns a list of Segments between detected scene changes.
+    """
+    try:
+        import ffmpeg
+    except ImportError:
+        raise ValueError(
+            "Couldn't import ffmpeg. Please make sure to "
+            "`pip install ffmpeg-python` explicitly or install "
+            "the correct extras like `pip install rikai[all]`"
+        )
+    assert isinstance(
+        video, (YouTubeVideo, VideoStream)
+    ), "Input type must be YouTubeVideo or VideoStream"
+
+    def isfloat(value):
+        try:
+            float(value)
+            return True
+        except ValueError:
+            return False
+
+    video_uri = (
+        video.get_stream().uri
+        if isinstance(video, YouTubeVideo)
+        else video.uri
+    )
+
+    p = subprocess.Popen(
```

Comment:
if we're going to start adding multiple ffmpeg related UDFs, we should think about adding a fluent API. Something that might look something like:

```
result = rikai.video.ffmpeg.run_pipline(
    ffmpeg.input(video_uri)
    .filter("select", "gte(scene,{})".format(threshold))
    .filter("showinfo")
    .output("-", format="null")
)
```

some of the type conversion tools can also be common utilities since all ffmpeg pipelines need to interact the cmdline subprocess